things/service.go

```diff
@@ -213,6 +243,10 @@ func (ts *thingsService) ViewThing(ctx context.Context, token, id string) (Thing
 		return Thing{}, errors.Wrap(ErrUnauthorizedAccess, err)
 	}
 
+	if err := ts.authorize(ctx, res.GetId(), id, ownerRelationKey, accessRelationKey); err != nil {
+		return Thing{}, err
+	}
+
 	return ts.things.RetrieveByID(ctx, res.GetEmail(), id)
 }
 
```

Comment:
do we need authorize in `ListThings`

Reply:
I've thought only admin fetches all things. However, for the sake of this PR, we only have ownership-based access. Therefore, we do not need to authorize here.

Reply:
User should be able to fetch all of the things he owns, and authorization is on the Keto ( though it is duplicated for now in the service itself)

Reply:
Authorization of list requests is very tricky, so I don't mind if we have simple filtering on the service layer in the initial version. We need to authenticate the user, though.